packages/flutter/lib/src/rendering/object.dart

```diff
@@ -756,18 +756,65 @@ class _ForkingSemanticsFragment extends _SemanticsFragment {
   }
 }
 
+/// The pipeline owner drives the rendering pipeline.
+///
+/// Every time the framework needs to produce a frame, the pipeline owner
+/// flushes any pending updates in the render tree.
```

Comment:
Actually technically that's what the binding does. This object provides the hooks to drive the pipeline, but the binding is the one that actually calls each step in turn.